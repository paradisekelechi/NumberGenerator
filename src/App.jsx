import React, { Component } from 'react';
import ReactToPrint from "react-to-print";
import { generateNewNumber, getNumberList, clearStorageList } from './utils/generator';
import PDFPrint from './PDFPrint';
import './App.scss';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      total: 0,
      number: '',
      isGenerating: false
    }
  }

  componentDidMount() {
    const numberList = getNumberList();
    if (numberList) {
      this.setState({
        total: numberList.length
      })
    }
  }

  generateNumber = () => {
    const generatedNumber = generateNewNumber();
    const numberList = getNumberList();
    if (numberList) {
      this.setState({
        number: generatedNumber,
        total: numberList.length
      })
    } else {
      this.setState({
        number: generatedNumber,
        total: 0
      });
    }
  }

  clearStorage = () => {
    clearStorageList();
    window.location.reload();
  }

  render() {
    const { total, number } = this.state;
    return (
      <div className="App">
        <div className="content">
          <div className="stat">
            <div className="value">{total}</div>
            <div className="title">Generated Numbers</div>
          </div>
          <div className="main">
            <div className="header">
              <h1>{number === '' ? ('. . . .') : number}</h1>
            </div>
            <div className="button-wrapper">
              <button onClick={this.generateNumber} className="button generate">Generate New Number</button>
              <ReactToPrint
                trigger={() => <button className="button download">Download Generated Numbers</button>}
                content={() => this.componentRef}
                onAfterPrint={this.clearStorage}
              />
              <div style={{ display: 'none' }}>
                <PDFPrint ref={el => (this.componentRef = el)} />
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default App;
