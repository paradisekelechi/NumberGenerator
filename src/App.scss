@import url('https://fonts.googleapis.com/css?family=Coiny');

$base-font: 'Coiny', cursive;

body, html, #root{
  height: 100%;
  width: 100%;
}
.App {
  text-align: center;
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}

.content{
  padding: 20px;
  width: 100%;
  box-sizing: border-box;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  font-family: $base-font;

  .stat{
    width: 20%;
    height: 270px;
    background-color: #000;
    color: #fff;
    border-radius: 20px;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    flex-wrap: wrap;

    .value{
      width: 100%;
      font-size: 5em;
    }
    .title{
      width: 100%;
      font-size: 2em;
    }
  }
  .main{
    width: 55%;
    height: 270px;

    .header{
      width: 100%;
      height: 122px;
      border-radius: 20px;
      border: 1px solid #000;
      background-color: white;
      display: flex;
      justify-content: center;
      align-items: center;
      text-align: center;
      h1{
        font-size: 5em;
      }
    }
    .button-wrapper{
      display: flex;
      align-items: flex-end;
      justify-content: space-between;
      height: 150px;
      button{
        width: 43%;
        height: 66px;
        border-radius: 20px;
        font-family: $base-font;
        font-size: 1.2em;
        border: none;
        box-shadow: 4px 4px 22px 5px rgba(168,166,168,1);
        cursor: pointer;
      }
      .download{
        background-color: #000;
        color: #fff;
      }
    }
  }
}


@media screen and (max-width: 980px) {
  .content{
    .stat{
      .value{
        font-size: 3em;
      }
      .title{
        font-size: 1em;
      }
    }
    .main{
      .header{
        h1{
          font-size: 3em;
        }
      }
      .button-wrapper{
        display: flex;
        align-items: flex-end;
        justify-content: space-between;
        height: 150px;
        flex-wrap: wrap;
        button{
          width: 100%;
          height: 66px;
          border-radius: 20px;
          font-family: $base-font;
          font-size: 1.2em;
          border: none;
          box-shadow: 4px 4px 22px 5px rgba(168,166,168,1);
          cursor: pointer;
        }
        .download{
          background-color: #000;
          color: #fff;
        }
      }
    }
  }
}


@media screen and (max-width: 680px) {
  .content{
    padding: 20px;
    width: 100%;
    box-sizing: border-box;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-evenly;
    font-family: $base-font;

    .stat{
      width: 100%;
      height: 200px;

      .value{
        width: 100%;
        font-size: 2.5em;
      }
      .title{
        width: 100%;
        font-size: 1em;
      }
    }
    .main{
      width: 100%;
      height: 270px;
      margin-top: 20px;

      .header{
        width: 100%;
        height: 122px;

        h1{
          font-size: 2.5em;
        }
      }
      .button-wrapper{
        display: flex;
        align-items: flex-end;
        justify-content: space-between;
        height: 150px;
        button{
          width: 100%;
        }
        .download{
          background-color: #000;
          color: #fff;
        }
      }
    }
  }
}
