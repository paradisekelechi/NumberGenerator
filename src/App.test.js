import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import './setupTests';
import App from './App';

it('renders without crashing', () => {
  const wrapper = shallow(<App />);
  const titleDiv = <div className="title">Generated Numbers</div>;
  expect(wrapper.contains(titleDiv)).toEqual(true);
});

it('should match snapshot', () => {
  const wrapper = shallow(<App />);
  console.log(wrapper.debug());
  expect(wrapper).toMatchSnapshot();
});

it('should contain the title', () => {
  const wrapper = shallow(<App />);
  const titleDiv = <div className="title">Generated Numbers</div>;
  expect(wrapper.contains(titleDiv)).toEqual(true);
  expect(wrapper.find('.title')).toBeDefined();
  expect(wrapper.find('ReactToPrint')).toBeDefined();
});

it('should contain the titlhhe', () => {
  const wrapper = shallow(<App />);
  wrapper.setState({
    number: '0809098987',
    total: 1
  });
  console.log(wrapper.debug());
  const header = <h1>0809098987</h1>;
  expect(wrapper.contains(header)).toEqual(true);
});




