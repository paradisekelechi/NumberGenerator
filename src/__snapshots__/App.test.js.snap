// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="content"
      >
        <div
          className="stat"
        >
          <div
            className="value"
          >
            0
          </div>
          <div
            className="title"
          >
            Generated Numbers
          </div>
        </div>
        <div
          className="main"
        >
          <div
            className="header"
          >
            <h1>
              . . . .
            </h1>
          </div>
          <div
            className="button-wrapper"
          >
            <button
              className="button generate"
              onClick={[Function]}
            >
              Generate New Number
            </button>
            <t
              bodyClass=""
              content={[Function]}
              copyStyles={true}
              onAfterPrint={[Function]}
              trigger={[Function]}
            />
            <div
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <PDFPrint />
            </div>
          </div>
        </div>
      </div>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="stat"
          >
            <div
              className="value"
            >
              0
            </div>
            <div
              className="title"
            >
              Generated Numbers
            </div>
          </div>,
          <div
            className="main"
          >
            <div
              className="header"
            >
              <h1>
                . . . .
              </h1>
            </div>
            <div
              className="button-wrapper"
            >
              <button
                className="button generate"
                onClick={[Function]}
              >
                Generate New Number
              </button>
              <t
                bodyClass=""
                content={[Function]}
                copyStyles={true}
                onAfterPrint={[Function]}
                trigger={[Function]}
              />
              <div
                style={
                  Object {
                    "display": "none",
                  }
                }
              >
                <PDFPrint />
              </div>
            </div>
          </div>,
        ],
        "className": "content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="value"
              >
                0
              </div>,
              <div
                className="title"
              >
                Generated Numbers
              </div>,
            ],
            "className": "stat",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 0,
                "className": "value",
              },
              "ref": null,
              "rendered": 0,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Generated Numbers",
                "className": "title",
              },
              "ref": null,
              "rendered": "Generated Numbers",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="header"
              >
                <h1>
                  . . . .
                </h1>
              </div>,
              <div
                className="button-wrapper"
              >
                <button
                  className="button generate"
                  onClick={[Function]}
                >
                  Generate New Number
                </button>
                <t
                  bodyClass=""
                  content={[Function]}
                  copyStyles={true}
                  onAfterPrint={[Function]}
                  trigger={[Function]}
                />
                <div
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                >
                  <PDFPrint />
                </div>
              </div>,
            ],
            "className": "main",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1>
                  . . . .
                </h1>,
                "className": "header",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": ". . . .",
                },
                "ref": null,
                "rendered": ". . . .",
                "type": "h1",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="button generate"
                    onClick={[Function]}
                  >
                    Generate New Number
                  </button>,
                  <t
                    bodyClass=""
                    content={[Function]}
                    copyStyles={true}
                    onAfterPrint={[Function]}
                    trigger={[Function]}
                  />,
                  <div
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                  >
                    <PDFPrint />
                  </div>,
                ],
                "className": "button-wrapper",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Generate New Number",
                    "className": "button generate",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Generate New Number",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bodyClass": "",
                    "content": [Function],
                    "copyStyles": true,
                    "onAfterPrint": [Function],
                    "onBeforePrint": undefined,
                    "pageStyle": undefined,
                    "trigger": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <PDFPrint />,
                    "style": Object {
                      "display": "none",
                    },
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {},
                    "ref": [Function],
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="content"
        >
          <div
            className="stat"
          >
            <div
              className="value"
            >
              0
            </div>
            <div
              className="title"
            >
              Generated Numbers
            </div>
          </div>
          <div
            className="main"
          >
            <div
              className="header"
            >
              <h1>
                . . . .
              </h1>
            </div>
            <div
              className="button-wrapper"
            >
              <button
                className="button generate"
                onClick={[Function]}
              >
                Generate New Number
              </button>
              <t
                bodyClass=""
                content={[Function]}
                copyStyles={true}
                onAfterPrint={[Function]}
                trigger={[Function]}
              />
              <div
                style={
                  Object {
                    "display": "none",
                  }
                }
              >
                <PDFPrint />
              </div>
            </div>
          </div>
        </div>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="stat"
            >
              <div
                className="value"
              >
                0
              </div>
              <div
                className="title"
              >
                Generated Numbers
              </div>
            </div>,
            <div
              className="main"
            >
              <div
                className="header"
              >
                <h1>
                  . . . .
                </h1>
              </div>
              <div
                className="button-wrapper"
              >
                <button
                  className="button generate"
                  onClick={[Function]}
                >
                  Generate New Number
                </button>
                <t
                  bodyClass=""
                  content={[Function]}
                  copyStyles={true}
                  onAfterPrint={[Function]}
                  trigger={[Function]}
                />
                <div
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                >
                  <PDFPrint />
                </div>
              </div>
            </div>,
          ],
          "className": "content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="value"
                >
                  0
                </div>,
                <div
                  className="title"
                >
                  Generated Numbers
                </div>,
              ],
              "className": "stat",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 0,
                  "className": "value",
                },
                "ref": null,
                "rendered": 0,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Generated Numbers",
                  "className": "title",
                },
                "ref": null,
                "rendered": "Generated Numbers",
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="header"
                >
                  <h1>
                    . . . .
                  </h1>
                </div>,
                <div
                  className="button-wrapper"
                >
                  <button
                    className="button generate"
                    onClick={[Function]}
                  >
                    Generate New Number
                  </button>
                  <t
                    bodyClass=""
                    content={[Function]}
                    copyStyles={true}
                    onAfterPrint={[Function]}
                    trigger={[Function]}
                  />
                  <div
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                  >
                    <PDFPrint />
                  </div>
                </div>,
              ],
              "className": "main",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h1>
                    . . . .
                  </h1>,
                  "className": "header",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": ". . . .",
                  },
                  "ref": null,
                  "rendered": ". . . .",
                  "type": "h1",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <button
                      className="button generate"
                      onClick={[Function]}
                    >
                      Generate New Number
                    </button>,
                    <t
                      bodyClass=""
                      content={[Function]}
                      copyStyles={true}
                      onAfterPrint={[Function]}
                      trigger={[Function]}
                    />,
                    <div
                      style={
                        Object {
                          "display": "none",
                        }
                      }
                    >
                      <PDFPrint />
                    </div>,
                  ],
                  "className": "button-wrapper",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Generate New Number",
                      "className": "button generate",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Generate New Number",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "bodyClass": "",
                      "content": [Function],
                      "copyStyles": true,
                      "onAfterPrint": [Function],
                      "onBeforePrint": undefined,
                      "pageStyle": undefined,
                      "trigger": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <PDFPrint />,
                      "style": Object {
                        "display": "none",
                      },
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {},
                      "ref": [Function],
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
